tracer/src/Datadog.Tracer.Native/cor_profiler.cpp

```diff
@@ -215,6 +215,29 @@ HRESULT STDMETHODCALLTYPE CorProfiler::Initialize(IUnknown* cor_profiler_info_un
         return E_FAIL;
     }
 
+    WSTRING runtimeType = runtime_information_.is_core() ? (runtime_information_.major_version > 4 ? WStr(".NET") : WStr(".NET Core")) : WStr(".NET Framework");
+    Logger::Info("Runtime Information: ", runtimeType, " ", runtime_information_.major_version, ".", runtime_information_.minor_version, ".", runtime_information_.build_version);
+
+    // Check if we have to disable tiered compilation (due to https://github.com/dotnet/runtime/issues/77973)
+    bool disableTieredCompilation = false;
+    bool internal_workaround_77973_enabled = true;
+    shared::TryParseBooleanEnvironmentValue(shared::GetEnvironmentValue(environment::internal_workaround_77973_enabled), internal_workaround_77973_enabled);
+    if (internal_workaround_77973_enabled)
+    {
+        if (runtime_information_.major_version == 5 ||
+            (runtime_information_.major_version == 6 && runtime_information_.minor_version == 0 && runtime_information_.build_version <= 11) ||
+            (runtime_information_.major_version == 7 && runtime_information_.minor_version == 0 && runtime_information_.build_version == 0))
+        {
+            Logger::Warn("Tiered Compilation was disabled due to https://github.com/dotnet/runtime/issues/77973. This action can be disabled by setting the environment variable DD_INTERNAL_WORKAROUND_77973_ENABLED=false");
+            disableTieredCompilation = true;
+        }
+    }
+
+    if (runtime_information_.major_version == 6 && runtime_information_.minor_version == 0 && runtime_information_.build_version <= 11)
+    {
+        disableTieredCompilation = true;
+    }
```

Comment:
I think this is an accidental duplicate?
```suggestion
```

Reply:
yes it was already removed in [7893b40](https://github.com/DataDog/dd-trace-dotnet/pull/3506/commits/7893b4038f1bd9784703d18623e87bec851210fc)